docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+- **Common**: Contains common classes used by other components.
+
+### UI
+
+### Parser
+The Parser interface uses a series of classes to implement the various commands.
+![Parser_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParserClass.png)
+
+### Command
+The `commands` package includes the command pattern used in the application to handle user operations. An abstract `Command` class serves as the base for all specific command implementations. Each command class extends `Command` and overrides the `execute()` method to perform its intended action.
+
+![Command_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/CommandClass.png)
+
+### Data
+#### task
+The `task` package manages all task-related functionality. A `Task` class serves as the base for other task types: `Todo`, `Deadline`, and `Repeat`. Each task type extends `Task` and introduces additional attributes relevant to its behavior. The `TaskList` class maintains a collection of tasks and provides methods to add, delete, find and track the completion rate of tasks.
+
+![Task_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/TaskClassDiagram.png)
+### State Manager
+
+### Storage
+
+![Storage_Class_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StorageClassDiagram.png)
+
+**API** : [`StorageFile.java`](https://github.com/AY2425S1-CS2113-T11-1/tp/tree/master/src/main/java/seedu/duke/storage/StorageFile.java)
+
+The `Storage` component,
+
+- Can save hospital data in Json format to a file, and read it back into hospital class.
+- Depends on the `Hospital` class in `Data` component as `Storage` component's job to save retrieve objects that belongs to `Data` gracefully.
+
+### Common Classes
+
+Classes that are used by multiple components are placed in the `Common` package.
+
+
+## Implementation
+
+### **AddPatientCommand**
+
+The add patient feature allows users to add a new patient to the hospital system. This is facilitated by the `AddPatientCommand`, which handles the logic of adding a patient and updating the system’s state and storage.
 
-## Design & implementation
+1. **User Input**: The user enters the `addPatient` command followed by patient details (e.g., name).
+2. **Command Parsing**: The `Parser` parses the input and creates an `AddPatientCommand` object.
+3. **Execution**: The `AddPatientCommand` checks if the patient already exists in the `Hospital`. If not, it adds the patient.
+4. **Storage Update**: The updated hospital data, now containing the new patient, is saved to storage.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Sequence Diagram
 
+The following sequence diagram illustrates how the `AddPatientCommand` is executed:
+
+![AddPatientCommand Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddPatientSequenceDiagram.png)
```

Comment:
Good sequence diagram, helps to make the flow of the program more understandable.